drivers/sensor/st/lis2dux12/st_lis2dux12_api.c

```diff
@@ -0,0 +1,238 @@
+	}
+
+	if (lis2dux12->data_ready_handler != NULL) {
+		lis2dux12->data_ready_handler(dev, lis2dux12->data_ready_trigger);
+	}
+
+exit:
+	ret = gpio_pin_interrupt_configure_dt(lis2dux12->drdy_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret < 0) {
+		LOG_ERR("%s: Not able to configure pin_int", dev->name);
+	}
+}
+
+static int32_t st_lis2dux12_init_interrupt(const struct device *dev)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_pin_int_route_t route;
+	int err;
+
+	/* Enable pulsed mode */
+	err = lis2dux12_data_ready_mode_set(ctx, LIS2DUX12_DRDY_PULSED);
+	if (err < 0) {
+		return err;
+	}
+
+	/* route data-ready interrupt on int1 */
+	err = lis2dux12_pin_int1_route_get(ctx, &route);
+	if (err < 0) {
+		return err;
+	}
+
+	route.drdy = 1;
+
+	err = lis2dux12_pin_int1_route_set(ctx, &route);
+	if (err < 0) {
+		return err;
+	}
+
+	return 0;
+}
+#endif
+
+const struct lis2dux12_chip_api st_lis2dux12_chip_api = {
+	.set_odr_raw = st_lis2dux12_set_odr_raw,
+	.set_range = st_lis2dux12_set_range,
+	.sample_fetch_accel = st_lis2dux12_sample_fetch_accel,
+#ifdef CONFIG_LIS2DUX12_ENABLE_TEMP
+	.sample_fetch_temp = st_lis2dux12_sample_fetch_temp,
+#endif
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+	.handle_interrupt = st_lis2dux12_handle_interrupt,
+	.init_interrupt = st_lis2dux12_init_interrupt,
+#endif
+};
+
+int st_lis2dux12_init(const struct device *dev)
+{
+	const struct lis2dux12_config *const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	uint8_t chip_id;
+	int ret;
+
+	lis2dux12_exit_deep_power_down(ctx);
+	k_busy_wait(25000);
+
+	/* check chip ID */
+	ret = lis2dux12_device_id_get(ctx, &chip_id);
+	if (ret < 0) {
+		LOG_ERR("%s: Not able to read dev id", dev->name);
+		return ret;
+	}
+
+	if (chip_id != LIS2DUX12_ID) {
+		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, chip_id);
+		return -EINVAL;
+	}
+
+	/* reset device */
+	ret = lis2dux12_init_set(ctx, LIS2DUX12_RESET);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_busy_wait(100);
+
+	LOG_INF("%s: chip id 0x%x", dev->name, chip_id);
+
+	/* Set bdu and if_inc recommended for driver usage */
+	lis2dux12_init_set(ctx, LIS2DUX12_SENSOR_ONLY_ON);
+
+	lis2dux12_timestamp_set(ctx, PROPERTY_ENABLE);
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+	if (cfg->trig_enabled) {
+		ret = lis2dux12_trigger_init(dev);
+		if (ret < 0) {
+			LOG_ERR("%s: Failed to initialize triggers", dev->name);
+			return ret;
+		}
+	}
+#endif
+
+	/* set sensor default pm and odr */
+	LOG_DBG("%s: pm: %d, odr: %d", dev->name, cfg->pm, cfg->odr);
+	ret = st_lis2dux12_set_odr_raw(dev, cfg->odr);
+	if (ret < 0) {
+		LOG_ERR("%s: odr init error (12.5 Hz)", dev->name);
+		return ret;
+	}
+
+	/* set sensor default scale (used to convert sample values) */
+	LOG_DBG("%s: range is %d", dev->name, cfg->range);
+	ret = st_lis2dux12_set_range(dev, cfg->range);
+	if (ret < 0) {
+		LOG_ERR("%s: range init error %d", dev->name, cfg->range);
+		return ret;
+	}
+
+	return 0;
+}
```

Comment:
> The driver specific api files are all prefixed with `st_`. From the folder path, users know already that it is a ST driver, is this prefix required?

It is because the compat string contains it, e.g. ```st_lis2dux12```

Reply:
Ok, so with the prefix you want to emphasize that it's a device specific file and distinguish it from the generic files that do not have the prefix. I haven't seen many sensor drivers that use a vendor prefix in the file name, but saw now that some are using it (e.g. ti_hdc).

I am fine with both variants (with or without the `st_` prefix.

Reply:
OK, I'm gonna change the filenames into ```lis2dux12_api.c``` and ```lis2duxs12_api.c``` but I'm keeping the API names as ```st_lis2dux12_xxx()```. That's because the compatible string suggests this.